bin/nanocld/src/services/process/stats.rs

```diff
@@ -63,3 +63,43 @@ pub async fn stats_processes(
       ),
   )
 }
+
+/// Get stats of all processes of key name
+#[cfg_attr(feature = "dev", utoipa::path(
+  get,
+  tag = "Processes",
+  path = "/processes/{name}/stats",
+  params(
+    ("name" = String, Path, description = "Name of the process", example = "deploy-example")
+  ),
+  responses(
+    (status = 200, description = "Process stats", content_type = "application/vdn.nanocl.raw-stream", body = ProcessStats),
+    (status = 404, description = "Process does not exist", body = crate::services::openapi::ApiError),
+  )
+))]
+#[web::get("/processes/{name}/stats")]
+pub async fn processes_stats_by_name(
+  state: web::types::State<SystemState>,
+  path: web::types::Path<(String, String)>,
+) -> HttpResult<web::HttpResponse> {
+  let (_, process) = path.into_inner();
+  let stream =
+    state
+      .inner
+      .docker_api
+      .stats(process.as_str(), None)
```

Comment:
We need to be able to pass the options, you can reuse the `ProcessStatsQuery` structure, also need to specify them in the openapi spec

Reply:
You don't need as_str() i think just & is enough

Reply:
What other option should we accept? ProcessStatsQuery contain namespace which is irrelevant in `process_stats_by_name`.

Reply:
> You don't need as_str() i think just & is enough

Fixed it

Reply:
The same as the cargo stats endpoint but without the namespace, creating it's own structure will be too much we can reuse the one that already exists and just not document the namespace in the openapi spec which is the macro on top of the endpoint

Reply:
Used ProcessStatsQuery